### ISSUE TITLE
Multiple facetDistribution for one attribute

### ISSUE THREAD

**Issue by Atmden** [meilisearch/meilisearch#3134]
I have an index with attribute '**some_attribute**' for searching.
The attribute type is an array.
The number of entries in the attribute array is up to 1000.
When I make a request to get facets on this attribute, 

```
curl --location \
--request GET 'http://localhost:7700//indexes/_index_name_/search?limit=0&facets=some_attribute,other_attribute' \
--header 'Authorization: Bearer Masterkey'
```

I get several arrays with results.
Similar to

```
"facetDistribution": {
    "some_attribute": {},
    "some_attribute.0": {},
    "some_attribute.1": {},
    "some_attribute.2": {},
    ...
    "some_attribute.170": {},
    "other_attribute": {},
}
```
On another attribute '**other_attribute**', also of type array - I get one array as expected.

What determines this behavior and how to prevent getting multiple arrays for one facet?

**Meilisearch version: 0.29.1**

**Additional context**
OS Ubuntu 22.04

**Comment by oluademola** [comment]
Hello @Atmden 

That is strange. Could you please try with the latest version? Let me know if the issue persists. 

Thanks

**Comment by Atmden** [comment]
Hello @oluademola
Thanks for the answer.
I updated the version of Meilisearch on the server to v0.30.0
Deleted the old index - date.ms folder
Created a new index using:
**Laravel Scout v9.4.10
meilisearch/meilisearch-php - v0.24.2**
Index settings are standard:
```
maxValuesPerFacet - 100
maxTotalHits - 1000
```
Meilisearch is launched through the systemd service

```
[Service]
ExecStart=/usr/bin/meilisearch --http-addr 0.0.0.0:7700 --env=development --master-key=....... --no-analytics
Restart=always
```

But the problem remained.
I also get multiple facetDistribution for one of the fields.

Request:
`curl --location --request GET 'http://localhost:7700//indexes/clinics/search?facets=disease_slugs,procedure_slugs' --header 'Authorization: Bearer .......'`

Result from Postmat:
[https://drive.google.com/file/d/1xtD20xQpYcaUwJXPjbrEfWPTSPUyH5u7/view?usp=sharing](https://drive.google.com/file/d/1xtD20xQpYcaUwJXPjbrEfWPTSPUyH5u7/view?usp=sharing)

I do not understand why and by what principle the data is divided into "chunks".

Also added another problem.
In the Meilisearch web interface when entering into the search bar "77569"
I get 2 entries where 77569 is the city_id attribute
But if I make a request for filtering by city_id = 77569 through Postmat, I get a null result.
city_id is specified as a filterable attribute.
Other requests to filter by city_id show the correct result.
For example filtering city_id = 77538 will return the correct result

**Comment by Atmden** [comment]
Found some peculiarity of the index.
With the same data, in one case, the attribute type is written as an array, in the other case, as jsion.
How to explicitly specify the data type for an attribute?

[https://drive.google.com/file/d/1xv-OYSqJK7f-ArVguJP5c0Yad2JszeqL/view?usp=sharing](https://drive.google.com/file/d/1xv-OYSqJK7f-ArVguJP5c0Yad2JszeqL/view?usp=sharing)

**Comment by oluademola** [comment]
Hello @Atmden,

We just released [v0.29.3](https://github.com/meilisearch/meilisearch/releases/tag/v0.29.3), which you are welcome to test. It should resolve the problem with the filter not returning the correct documents. We are currently working on bug fixes for v0.30.0, which should be available very soon.

Please let us know if it helps.

Thanks.

**Comment by curquiza** [comment]
I close this issue due to a lack of answers. Nothing is definitive of course, feel free to ask to re-open if needed 😊
